src/Util/ClassUtils.php

```diff
@@ -110,4 +117,31 @@ public static function generateProxyClassName($className, $proxyNamespace)
     {
         return rtrim($proxyNamespace, '\\') . '\\' . Proxy::MARKER . '\\' . ltrim($className, '\\');
     }
+
+    /**
+     * Check if the type is an enum type or type containing an enum type
+     *
+     * @param ReflectionNamedType|ReflectionUnionType|ReflectionIntersectionType|null $type
+     *
+     */
+    public static function containsEnumType($type): bool
+    {
+        if (PHP_VERSION_ID <= 80100 || $type === null) {
+            return false;
+        }
+
+        if ($type instanceof ReflectionUnionType || $type instanceof ReflectionIntersectionType) {
+            foreach ($type->getTypes() as $unionedType) {
+                if (static::containsEnumType($unionedType)) {
```

Comment:
I would use `self::` here. there is no reason to enable late static binding.

Reply:
Yes